### PR TITLE
Sync package optimisations

### PR DESCRIPTION
**Pull request by psyren89**
Cleaned and formatted the sync package.

### CODE REVIEW

**Comment by Ruxton** [comment]
Does this do anything other than change some wordings? merge away.

**Comment by psyren89** [comment]
Changes for clarity, use of modern Java features (e.g., enhanced for loop), and some refactoring to remove 7-level deep nested `if` statements.

File: fh-android-sdk/src/main/java/com/feedhenry/sdk/sync/FHSyncConfig.java [hunk 1]
```diff
@@ -44,224 +44,224 @@
...
      */
     public void setNotifyUpdateFailed(boolean pNotifyUpdateFailed) {
         this.mNotifyRemoteUpdateFailed = pNotifyUpdateFailed;
     }
 
     /**
-     * Does the sync client notify remote updates applied event
-     * 
-     * @return true or false
+     * Gets whether the sync client notifies on a remote updates applied event.
```
Comment (kevin-jackson) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/sync/FHSyncConfig.java]: update or updates applied? `isNotifyRemoteUpdateApplied()`

File: fh-android-sdk/src/main/java/com/feedhenry/sdk/sync/FHSyncDataset.java [hunk 2]
```diff
@@ -202,33 +186,35 @@ public void startSyncLoop() {
...
-                    public void fail(FHResponse pResponse) {
-                        // The AJAX call failed to complete succesfully, so the state of the current pending updates is unknown
-                        // Mark them as "crashed". The next time a syncLoop completets successfully, we will review the crashed
-                        // records to see if we can determine their current state.
-                        markInFlightAsCrashed();
-                        FHLog.e(LOG_TAG, "syncLoop failed : msg = " + pResponse.getErrorMessage(),
+                        @Override
+                        public void fail(FHResponse pResponse) {
+                            /*
+                            The AJAX call failed to complete succesfully, so the state of the current pending updates is
```
Comment (kevin-jackson) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/sync/FHSyncDataset.java]: *successfully

File: fh-android-sdk/src/main/java/com/feedhenry/sdk/sync/FHSyncDataset.java [hunk 3]
```diff
@@ -202,33 +186,35 @@ public void startSyncLoop() {
...
+                            unknown. Mark them as "crashed". The next time a syncLoop completets successfully, we will
```
Comment (kevin-jackson) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/sync/FHSyncDataset.java]: *completes

File: fh-android-sdk/src/main/java/com/feedhenry/sdk/sync/FHSyncDataset.java [hunk 4]
```diff
@@ -402,218 +387,266 @@ private void resetDataRecords(JSONObject pData) {
...
+
+                resolvedCrashed.put(uid, crashedUpdate);
+                FHLog.d(
+                    LOG_TAG,
+                    "updateCrashedInFlightFromNewData - Resolving status for crashed inflight pending record " +
+                        crashedUpdate);
+                String crashedType = crashedUpdate.optString("type", null);
+                String crashedAction = crashedUpdate.optString("action", null);
+                if ("failed".equalsIgnoreCase(crashedType)) {
+                    // Crashed updated failed - revert local dataset
```
Comment (kevin-jackson) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/sync/FHSyncDataset.java]: *update

**Pull request merged**